test/pinning_test.jl

```diff
@@ -66,3 +66,18 @@ end
     cpuids_after = getcpuids()
     @test check_compact_within_numa(cpuids_after)
 end
+
+@testset "Thread Pinning (firstn)" begin
+    pinthreads(sort(getcpuids(); rev=true))
+    @test issorted(getcpuids(); rev=true)
+
+    pinthreads(:firstn)
+    @test issorted(getcpuids())
+end
+
+@testset "Thread Pinning (random)" begin
+    pinthreads(:firstn)
+
+    pinthreads(:random)
+    @test !issorted(getcpuids())
```

Comment:
It could be sorted by chance! I guess what we could test is that if we do `pinthreads(:random)`, say, 10 times that `getcpuids()` must not be the same for all of the runs. (Again, although very unlikely, by chance it could be that a few runs actually give exactly the same cpuids but I would think that 10 is reasonably large to practically rule that out.)